.github/workflows/test.yml

```
@@ -123,3 +123,76 @@ jobs:
       - name: 'Tear down Docker'
         if: always()
         run: docker stop --time 0 mir-smir-ci-${GITHUB_SHA}
+
+  publish-kmir-image:
+    name: Publish Kmir Image
+    needs: version-bump
```

We should maybe not even try to build the image when any tests are failing, meaning we'd want this to require `unit-tests` and `integration-tests` (and `stable-mir-json-integration-tests` but that's probably less important).
```suggestion
    needs: [unit-tests, integration-tests]
```
With this change, we _could_ push the PR images to `ghcr.io` but I would still recommend not pushing.